packages-exp/auth-exp/src/core/user/token_manager.ts

```diff
@@ -99,18 +100,47 @@ export class StsTokenManager {
     return null;
   }
 
-  // TODO(lisajian): Fill in once refresh listener is added
   private async getTokenAndTriggerCallback(
-    _auth: AuthInternal,
-    _forceRefresh = false
+    auth: AuthInternal,
+    forceRefresh = false
   ): Promise<string | null> {
-    return null;
+    if (forceRefresh) {
+      // TODO(lisajian): Add and use client error code - ERROR_TOKEN_REFRESH_UNAVAILABLE
+      _assert(auth._customTokenProvider, auth, AuthErrorCode.INTERNAL_ERROR);
+      return this.refreshWithCustomToken(auth);
+    }
+
+    // isExpired includes a buffer window, during which the access token will still be returned even
+    // if a customTokenProvider is not set yet.
+    if (this.isExpired) {
```

Comment:
you can have a lot less indentation if you swap things around here a bit and change this if to `if(!this.isExpired) { return this.accessToken; }`